control/xferfcn.py

```diff
@@ -930,11 +937,28 @@ def _common_den(self, imag_tol=None):
                         nwzeros.append(poles[j][ip])
 
                     numpoly = poleset[i][j][2] * np.atleast_1d(poly(nwzeros))
+
+                    # td04ad expects a proper transfer function. If the
+                    # numerater has a higher order than the denominator, the
+                    # padding will fail
+                    if len(numpoly) > maxindex + 1:
+                        if allownonproper:
+                            havenonproper = True
```

Comment:
Why not simply set num = None, and break here?
Don't see the need for the havenonproper flag.

Reply:
The outer loop for creating the denominator needs to finish for all inputs. If we just break, we might try to assign a `numpoly` to `num[i, j, ...]` on the next iteration which fails because `None` cannot be sliced.